### PR TITLE
Bft verify signature

### PR DESCRIPTION
**Pull request by danielSanchezQ**

### CODE REVIEW

**Comment by danielSanchezQ** [comment]
@dkijania We should include a test for this (?)

File: chain-impl-mockchain/src/leadership/bft.rs [hunk 1]
```diff
@@ -45,13 +45,29 @@ impl LeadershipData {
...
+                let leader = self.get_leader_at(block_header.block_date());
+
+                if leader_id != &leader {
                     Verification::Failure(Error::new(ErrorKind::InvalidLeader))
                 } else {
-                    Verification::Success
+                    // verify block signature
+                    match signature
+                        .0
+                        .verify_slice(leader.as_public_key(), block_header.as_slice())
```
Comment (vincenthz) [inline chain-impl-mockchain/src/leadership/bft.rs:61]: you need to validate: 
```
block_header.as_auth_slice();
```
not
```
block_header.as_slice();
```

Comment (danielSanchezQ) [reply chain-impl-mockchain/src/leadership/bft.rs:61]: Noted, Thanks @vincenthz!

**Review by mzabaluev — APPROVED** [review]
Looks good, CI is green, haven't seen a better PR in a while!

**Pull request merged**